### PR TITLE
fix(migrate): use the right node version (without >=)

### PR DESCRIPTION
**Pull request by clementprevot**

### CODE REVIEW

**Comment by clementprevot** [comment]
My previous fix for this was bugged because of a mis-interpretation of the node version to a redirect.

**Comment by Kikobeats** [comment]
@clementprevot I refactor the code. Can you merge master?

**Comment by clementprevot** [comment]
I saw that, I was merging the conflicts. You are way too fast for me :wink:

**Comment by Kikobeats** [comment]
haha, but you resolve the conflict using the previous version!

**Comment by clementprevot** [comment]
Really? Hum that's weird, let me check that

**Comment by clementprevot** [comment]
@Kikobeats Are you sure? When I look at the changes of the PR, I only see the changes to use the right node version (variable name `currentVersion` in both `index.js` and `switcher.js` files)

File: bin/index.js [hunk 1]
```diff
@@ -27,8 +27,10 @@ var rangeVersion = pkg.engines && pkg.engines.node
 if (!rangeVersion) processExit()
 
 config(function (versions) {
+  var currentVersion = process.versions.node
+
```
Comment (Kikobeats) [inline bin/index.js]: can you remove this extra line?

File: bin/index.js [hunk 2]
```diff
@@ -27,8 +27,10 @@ var rangeVersion = pkg.engines && pkg.engines.node
 if (!rangeVersion) processExit()
 
 config(function (versions) {
+  var currentVersion = process.versions.node
```
Comment (Kikobeats) [inline bin/index.js:30]: If you are tying to switch the version, this version refers the old version.

For sanetize the version, use `semver.sanetize`:

```
semver.clean('  =v1.2.3   ') // '1.2.3'
```

**Comment by Kikobeats** [comment]
@clementprevot lol you're right, I added comments 👍

File: bin/switcher.js [hunk 3]
```diff
@@ -2,20 +2,20 @@
 
 var spawn = require('child_process').spawn
```
Comment (Kikobeats) [inline bin/switcher.js:4]: `var semver = require('semver')` here

File: bin/switcher.js [hunk 4]
```diff
@@ -2,20 +2,20 @@
...
 
-function nvmCommand (maxSatisfyVersion, rangeVersion) {
+function nvmCommand (maxSatisfyVersion, currentVersion) {
   var cmd = ''
   cmd += 'nvm install '
   cmd += maxSatisfyVersion
```
Comment (Kikobeats) [inline bin/switcher.js:8]: `cmd += semver.clean(maxSatisfyVersion)` here

File: bin/index.js (hunk 1 above)
Comment (clementprevot) [reply bin/index.js]: Done

**Comment by Kikobeats** [comment]
Need to resolve the pending reviews before merge this

**Comment by clementprevot** [comment]
I don't see any pending review

**Review by Kikobeats — CHANGES_REQUESTED** [review]

**Comment by Kikobeats** [comment]
was (again) my fault, check now 😄

File: bin/index.js (hunk 2 above)
Comment (clementprevot) [reply bin/index.js:30]: No, I precisely want to get the current (so the old) version of node, to be able to tell to NVM to migrate already installed package from this version.

e.g: we are in node 6.7.0, NVM detects a new 6.8.1 version, I want to migrate packages from 6.7.0 so the command line is `nvm install 6.8.1 --resinstall-packages-from=6.7.0`. That's the main goal of the `currentVersion` param given to the `createSwitcher` function

File: bin/switcher.js (hunk 3 above)
Comment (clementprevot) [reply bin/switcher.js:4]: See my previous comment, thus, not needed

File: bin/switcher.js (hunk 4 above)
Comment (clementprevot) [reply bin/switcher.js:8]: See again my first comment

**Comment by clementprevot** [comment]
I commented on your reviews. I think you misunderstood what I wanted to do :wink:

**Review by Kikobeats — APPROVED** [review]

**Comment by Kikobeats** [comment]
yay!

**Pull request merged**